src/creds.jl

```diff
@@ -36,3 +48,16 @@ mutable struct ClientCredentials <: Credentials
     ClientCredentials(client_id, client_secret, client_credentials_url = nothing) =
         new(client_id, client_secret, client_credentials_url, nothing)
 end
+
+function Base.show(io::IO, c::ClientCredentials)
+    print(
+        io,
+        "(",
+        c.client_id,
+        c.client_secret == nothing ? "" : ", $(c.client_secret[1:3])...",
+        c.access_token == nothing ? "" : ", $(c.access_token)",
+        ", ",
+        c.client_credentials_url,
+        ")"
+    )
+end
```

Comment:
🤔 shouldn't you not need this? I think that it should automatically default to recursively calling show() on the inner arguments. For example:
```julia
julia> struct A
           x::Int
       end

julia> Base.show(io::IO, a::A) = print(io, "HELLO WORLD: $(a.x)")

julia> struct B
           a::A
           y::Int
       end

julia> B(A(1), 2)
B(HELLO WORLD: 1, 2)
```

Reply:
oh, i'm sorry never mind. I see now that you're also trying to hide the client_secret in this struct. 👍